test/libsolidity/smtCheckerTests/modifiers/modifier_code_after_placeholder.sol

```diff
@@ -21,5 +21,5 @@ contract C
 	}
 }
 // ----
-// Warning 4984: (203-208): Overflow (resulting value larger than 2**256 - 1) happens here
+// Warning 4984: (203-208): Overflow (resulting value larger than 2**256 - 1) happens here.
```

Comment:
I am curious about how this happens, when it has `.` and when not?

Reply:
When the SMT query is satisfiable, usually meaning "there is an overflow, assertion failure, etc", we want to get a counterexample for the failed property. The problem with doing that in CHC is that the solver applies a lot of preprocessing that make retrieving counterexamples impossible. So we do a second run (same query) but with solver preprocessing disabled. The problem is now much harder for the solver, so it might timeout or answer `unknown`. If the solver still says SAT, we print a counterexample, so the `.` is not there. If the problem without preprocessing was indeed too hard and the solver couldn't solve it, we still know that the problem is SAT, so we can report "overflow etc", but we don't have a counterexample, so the message ends there with `.`.

Reply:
Ah, and the reason this changed here in this PR is that the encoding changed a bit, and this sometimes triggers different solving behaviors in the solver.

Reply:
Thanks for clarification